resolwe_bio/processes/samtools/samtools_coverage.py

```diff
@@ -0,0 +1,312 @@
+            "memory": 16384,
+        },
+    }
+    category = "Samtools"
+    data_name = "Samtools coverage"
+    version = "1.0.0"
+    scheduling_class = SchedulingClass.BATCH
+
+    class Input:
+        """Input fields for SamtoolsCoverageMulti."""
+
+        bam = ListField(
+            DataField(data_type="alignment:bam"),
+            label="Input BAM files",
+            description="Select BAM file(s) for the analysis. Coverage information will "
+            "be calculated from the merged alignments.",
+        )
+        region = StringField(
+            label="Region",
+            description="Region can be specified as: RNAME:STARTPOS-ENDPOS "
+            "and all position coordinates are 1-based, where RNAME is the "
+            "name of the contig. If the input BAM file was generated by "
+            "General RNA-seq pipeline, you should use only chromosome "
+            "numbers to subset the input file, e.g. 3:30293-39103.",
+            required=False,
+        )
+
+        class AdvancedOptions:
+            """Advanced options."""
+
+            min_read_length = IntegerField(
+                label="Minimum read length",
+                required=False,
+                description="Ignore reads shorter than specified number of "
+                "base pairs.",
+            )
+            min_mq = IntegerField(
+                label="Minimum mapping quality",
+                required=False,
+                description="Minimum mapping quality for an alignment to be used.",
+            )
+            min_bq = IntegerField(
+                label="Minimum base quality",
+                required=False,
+                description="Minimum base quality for a base to be considered.",
+            )
+            excl_flags = ListField(
+                StringField(),
+                label="Filter flags",
+                default=["UNMAP", "SECONDARY", "QCFAIL", "DUP"],
+                description="Filter flags: skip reads with mask bits set. "
+                "Press ENTER after each flag.",
+            )
+            depth = IntegerField(
+                label="Maximum allowed coverage depth",
+                default=1000000,
+                description="If 0, depth is set to the maximum integer value "
+                "effectively removing any depth limit.",
+            )
+            no_header = BooleanField(
+                label="No header",
+                default=False,
+                description="Do not output header.",
+            )
+
+        advanced = GroupField(AdvancedOptions, label="Advanced options")
+
+    class Output:
+        """Output fields for SamtoolsCoverageMulti."""
+
+        table = FileField(label="Output coverage table")
+        build = StringField(label="Build")
+        species = StringField(label="Species")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+
+        output_name = "samtools_coverage.tsv"
+
+        if len(inputs.bam) > 1:
```

Comment:
Fail the process if the number of input BAMs is exactly 1. Direct to the single sample process instead.